### PR TITLE
Make tests runnable

### PR DESCRIPTION
**Pull request by foxbunny**
In order to make tests runnable, we have introduced the following changes:

- Created a requirements file which contains all the necessary dependencies
- Wrote a new setup module command that grabs the requirements and creates a local mirror using [pip2pi](https://github.com/wolever/pip2pi)
- Added tox.ini that modifies pip behavior to use the local mirror

13 out of 26 tests are currently failing, but imports have been adjusted so that they actually run instead of failing right at the import line.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -202,7 +228,8 @@ def run(self):
     cmdclass={
         'test': PyTest,
         'develop': Develop,
-        'sdist': Package,
+        'deps': LocalMirror,
```
Comment (goodbadwolf) [inline setup.py]: 'deps' should be renamed to localmirror

Comment (foxbunny) [inline setup.py]: Ok

File: setup.py [hunk 2]
```diff
@@ -35,12 +36,11 @@ def read(fname):
...
 
 def in_scriptdir(path):
-    return os.path.join(SCRIPTDIR, os.path.normpath(path))
+    return join(SCRIPTDIR, normpath(path))
 
 
 def rebuild_catalogs():
-    import subprocess
-    subprocess.call('python scripts/cmpmsgs.py librarian/locales', shell=True)
+    call('python scripts/cmpmsgs.py librarian/locales', shell=True)
```
Comment (integricho) [inline setup.py]: I think this is obsolete since librarian_core was added

File: tests/presentation/test_dashboard_plugin.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import mock
 
-import librarian_dashboard.dashboard as mod
+import librarian.presentation.dashboard.dashboard as mod
```
Comment (integricho) [inline tests/presentation/test_dashboard_plugin.py]: We should agree on whether we go with absolute or relative imports

File: setup.py (hunk 2 above)
Comment (foxbunny) [inline setup.py]: How do you rebuild catalogs now?

File: tests/presentation/test_dashboard_plugin.py (hunk 3 above)
Comment (foxbunny) [inline tests/presentation/test_dashboard_plugin.py]: This is in a test, though.

File: setup.py (hunk 2 above)
Comment (integricho) [inline setup.py]: to collect messages: `librarian --conf librarian.ini --xmsgs`
to compile messages: `librarian --conf librarian.ini --cmsgs`

Comment (foxbunny) [inline setup.py]: This is triggered right before `sdist`, among other things. I think it's not a bad thing to have, as tox will execute a `sdist` before running tests.

File: tests/presentation/test_dashboard_plugin.py (hunk 3 above)
Comment (integricho) [inline tests/presentation/test_dashboard_plugin.py]: Riiiight, didn't notice, cool, relative it is then :)

**Pull request merged**